src/mjolnir/ingest_transit.cc

```diff
@@ -389,33 +390,50 @@ bool write_stop_pair(Transit& tile,
   const std::string currFeedPath = feed_trip.feed;
 
-  auto currTrip = feed.get_trip(tile_tripId);
+  const auto& currTrip = feed.get_trip(tile_tripId);
+  const auto& currShape = feed.get_shape(currTrip->shape_id);
+  const auto& trip_calendar = feed.get_calendar(currTrip->service_id);
 
   // already sorted by stop_sequence
   const auto tile_stopTimes = feed.get_stop_times_for_trip(tile_tripId);
 
   for (size_t stop_sequence = 0; stop_sequence < tile_stopTimes.size() - 1; stop_sequence++) {
-    gtfs::StopTime origin_stopTime = tile_stopTimes[stop_sequence];
-    gtfs::Id origin_stopId = origin_stopTime.stop_id;
-    auto origin_stop = feed.get_stop(origin_stopId);
+    const auto& origin_stopTime = tile_stopTimes[stop_sequence];
+    const auto& origin_stopId = origin_stopTime.stop_id;
+    const auto& origin_stop = feed.get_stop(origin_stopId);
     assert(origin_stop);
-    gtfs::StopTime dest_stopTime = tile_stopTimes[stop_sequence + 1];
-    gtfs::Id dest_stopId = dest_stopTime.stop_id;
-    auto dest_stop = feed.get_stop(dest_stopId);
+    const auto& dest_stopTime = tile_stopTimes[stop_sequence + 1];
+    const auto& dest_stopId = dest_stopTime.stop_id;
+    const auto& dest_stop = feed.get_stop(dest_stopId);
     assert(dest_stop);
-    auto origin_graphid_it = platform_node_ids.find({origin_stopId, currFeedPath});
-    auto dest_graphid_it = platform_node_ids.find({dest_stopId, currFeedPath});
+    const auto origin_graphid_it = platform_node_ids.find({origin_stopId, currFeedPath});
+    const auto dest_graphid_it = platform_node_ids.find({dest_stopId, currFeedPath});
     const bool origin_is_in_tile = origin_graphid_it != platform_node_ids.end();
     const bool dest_is_in_tile = dest_graphid_it != platform_node_ids.end();
 
     // check if this stop_pair (the origin of the pair) is inside the current tile
     if ((origin_is_in_tile || dest_is_in_tile) && origin_stopTime.trip_id == dest_stopTime.trip_id) {
+      auto* stop_pair = tile.add_stop_pairs();
 
-      dangles = dangles || !origin_is_in_tile || !dest_is_in_tile;
+      auto* service_dow = stop_pair->mutable_service_days_of_week();
+      service_dow->Add(trip_calendar->monday == gtfs::CalendarAvailability::Available);
+      service_dow->Add(trip_calendar->tuesday == gtfs::CalendarAvailability::Available);
+      service_dow->Add(trip_calendar->wednesday == gtfs::CalendarAvailability::Available);
+      service_dow->Add(trip_calendar->thursday == gtfs::CalendarAvailability::Available);
+      service_dow->Add(trip_calendar->friday == gtfs::CalendarAvailability::Available);
+      service_dow->Add(trip_calendar->saturday == gtfs::CalendarAvailability::Available);
+      service_dow->Add(trip_calendar->sunday == gtfs::CalendarAvailability::Available);
+      // skip if no days are valid
+      if (!trip_calendar || !service_dow->size()) {
+        LOG_WARN("Service ID " + currTrip->service_id + " has no calendar entry, skip.");
+        continue;
+      }
```

Comment:
tentative: don't think we need to care about this stop pair when there's no valid days in the week. but gotta validate that this is fine.